pkg/gofr/handler.go

```diff
@@ -46,10 +45,8 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if websocket.IsWebSocketUpgrade(r) {
 		// If the request is a WebSocket upgrade, do not apply the timeout
 		c.Context = r.Context()
-	} else if h.requestTimeout != "" {
-		reqTimeout := h.setContextTimeout(h.requestTimeout)
-
-		ctx, cancel := context.WithTimeout(r.Context(), time.Duration(reqTimeout)*time.Second)
+	} else if h.requestTimeout != 0 {
+		ctx, cancel := context.WithTimeout(r.Context(), h.requestTimeout*time.Second)
```

Comment:
Here is why you weren't able to reduce the time


```suggestion
		ctx, cancel := context.WithTimeout(r.Context(), h.requestTimeout)
```

Here you virtually created a timeout equal to the number on nanoseconds multiplied by 1 s

Reply:
We are asking the user to give the value only not units:

```go
reqTimeout, err := strconv.Atoi(a.Config.Get("REQUEST_TIMEOUT"))
	if err != nil && a.Config.Get("REQUEST_TIMEOUT") != "" {
		a.container.Error("invalid value of config REQUEST_TIMEOUT. setting default value to 5 seconds.")
	}

	a.httpServer.router.Add(method, pattern, handler{
		function:       h,
		container:      a.container,
		requestTimeout: time.Duration(reqTimeout),
	})
```


If i remove time.Second in `h.requestTimeout*time.Second` and use time.ParseDuration in above code user will need to give units in config

Reply:
reqTimeout is an interface

We can expect user to be OK to have seconds via the config. And it's what they had.

Our issue is to speed up the tests

So your code should be using

reqTimeOut * time.Second 

When passing the duration to the config

Reply:
Sorry my bad got it